packages/loot-core/src/shared/arithmetic.test.ts

```diff
@@ -41,4 +42,52 @@ describe('arithmetic', () => {
   test('respects current number format', () => {
     expect(evalArithmetic('1,222.45')).toEqual(1222.45);
   });
+
+  for (let format of ['dot-comma', 'comma-dot', 'space-comma', 'space-dot']) {
+    function testFormatCompatibility(testFormat, fn) {
+      test(
+        'format is agnostic (' + format + ' supports ' + testFormat + ')',
+        () => {
+          setNumberFormat({ format, hideFraction: false });
+          fn();
+        },
+      );
+    }
+
+    testFormatCompatibility('dot-comma', () => {
+      expect(evalArithmetic('1.234,56')).toEqual(1234.56);
```

Comment:
Can you also add a test case like `1.234.567` to all of these cases to double check that integers are parsed correctly?

Reply:
They're not being parsed correctly, let me fix that.